src/controllers/abstract-controller.ts

```diff
@@ -13,36 +12,25 @@ interface ParsedErrors {
 export default abstract class AbstractController {
   private entity: Repository<any>;
 
-  private key: string;
-
-  protected checkId = async (req: Request, _res: Response, next: NextFunction): Promise<void> => {
-    try {
-      if (!this.entity) {
-        throw new CustomError('Before use checkId method, please specify used entity with setMainEntity method', 400);
-      }
+  protected async checkId<TData>(id: string): Promise<TData> {
```

Comment:
As you changed purpose of this method, would be better to change it name appropriately:

before it worked as common middleware to populate request, now it's called directly and returns instance

Reply:
Renamed